app/models/fact_value.rb

```diff
@@ -9,7 +9,7 @@ class FactValue < ActiveRecord::Base
 
   has_one :parent_fact_name, :through => :fact_name, :source => :parent
 
-  scoped_search :on => :value, :in_key=> :fact_name, :on_key=> :name, :rename => :facts, :complete_value => true, :only_explicit => true
+  scoped_search :on => :value, :in_key=> :fact_name, :on_key=> :name, :rename => :facts, :complete_value => true, :only_explicit => true, :ext_method => :search_cast_facts
   scoped_search :on => :value, :default_order => true
```

Comment:
Yes, I meant the above suggestion only for the cases where we have a name - for the value only search, just use `{:condition => cast_facts(..)}`